pkgs/by-name/ia/ianny/package.nix

```
@@ -0,0 +1,43 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, dbus
+, pkg-config
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ianny";
+  version = "0.1.0alpha.2";
+
+  src = fetchFromGitHub {
+    owner = "zefr0x";
+    repo = "ianny";
+    rev = "v${version}";
+    hash = "sha256-UEkT9aNUN71LavjpMC0qnyTAcCt6dDfJmR7peclEMnY=";
+  };
+
+  cargoHash = "sha256-ajoF0c6+/XfJrNa2phHbAeozdhduzc4BKvNuBDXS3qQ=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ dbus.dev ];
+
+  postPatch = ''
+    substituteInPlace src/main.rs \
+      --replace '/usr/share/locale' $out/share/locale
+  '';
+
+  preFixup = ''
+    mkdir -p $out/etc/xdg/autostart
+    mkdir -p $out/usr/share/local
+    cp io.github.zefr0x.ianny.desktop $out/etc/xdg/autostart
```

Are you copying the file `io.github.zefr0x.ianny.desktop` to the file `$out/etc/xdg/autostart` or are you copying the file `io.github.zefr0x.ianny.desktop` in the directory `$out/etc/xdg/autostart` ? It's a bit confusing.

To avoid that, I usually add a trailing `/` at the end, as such:

```suggestion
    cp io.github.zefr0x.ianny.desktop $out/etc/xdg/autostart/
```

---

Thanks!